File: Trabalhos Teóricos/tt02/Dona monica/DonaMonica.java
class DonaMonica{
    
    public static boolean isFim(String s){
        return(s.length() == 1 && s.charAt(0) == '0');
    }


    public static int calculaIdade(String linha) {
        String idadeMonica = "";
        String idadeFilho1 = "";
        String idadeFilho2 = "";
        int i = 0;
        while(linha.charAt(i) != ' '){
            idadeMonica += linha.charAt(i);
            i++;
        }i++;
        while(linha.charAt(i) != ' '){
            idadeFilho1 += linha.charAt(i);
            i++;
        }i++;
        while(i < linha.length()){
            idadeFilho2 += linha.charAt(i);
            i++;
        }

        int ageMonica;
        int idades[] = new int[3];

        ageMonica = Integer.parseInt(idadeMonica);
        idades[0] = Integer.parseInt(idadeFilho1);
        idades[1] = Integer.parseInt(idadeFilho2);

        idades[2] = ageMonica - (idades[0]+idades[1]);

        int maior = idades[2];
        for(int j = 0; j < 2; j++){
            if(idades[j] > maior){
                maior = idades[j];
            }
        }
        return maior;
    }

    public static void main(String[] args) {
        
        String[] entrada = new String[1000];
        int numEntrada = 0;

        do{
            entrada[numEntrada] = MyIO.readLine();
        }while(isFim(entrada[numEntrada++]) == false);
        numEntrada--;

        for(int i = 0; i < numEntrada; i++){
            MyIO.println(calculaIdade(entrada[i]));
        }

    }
}

File: Trabalhos Teóricos/tt02/Dona monica/DonaMonica.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>
#include <stdlib.h>

#define NUMENTRADA 1000
#define TAMLINHA   1000

bool isFim(char* s){
    return(strlen(s) >= 1 && s[0] == '0');
}

int calculaIdade(char* s){
    char idadeMonica[5];
    char idadeFilho1[5];
    char idadeFilho2[5];
    int j = 0;

    int i = 0;
    while(s[i] != ' '){
        idadeMonica[j] = s[i];
        i++;
        j++;
    }i++;
    j = 0;
    while(s[i] != ' '){
        idadeFilho1[j] = s[i];
        i++;
        j++;
    }i++;
    j=0;
    while(i < strlen(s)){
        idadeFilho2[j] = s[i];
        i++;
        j++;
    }

    int ageMonica;
    int idades[3]; 
    ageMonica = atoi(idadeMonica);
    idades[0] = atoi(idadeFilho1);
    idades[1] = atoi(idadeFilho2);

    idades[2] = ageMonica - (idades[0]+idades[1]);

    int maior = idades[2];
        for(int z = 0; z < 2; z++){
            if(idades[z] > maior){
                maior = idades[z];
            }
        }
    return maior;

}

int main(){

    // inicio leitura pub.in
    char linha[NUMENTRADA][TAMLINHA];
    int numEntrada = 0;

    do
    {
        fgets(linha[numEntrada], TAMLINHA, stdin);
    } while (isFim(linha[numEntrada++]) == false);
    numEntrada--;
    // fim leitura pub.in

    for(int i = 0; i < numEntrada; i++){
        printf("%d\n", calculaIdade(linha[i]));
    }

}